### PR TITLE
TAG: Tangential Amplifying Guidance

### PR DESCRIPTION
**Pull request by kabachuha**
This Pull Request implements "TAG: Tangential Amplifying Guidance for Hallucination-Resistant Diffusion Sampling" from https://huggingface.co/papers/2510.04533.

The code is organized in the same manner as the vanilla TCFG node with the similar, but different concept.

Tangential Amplifying Guidance (TAG) improves diffusion model sample quality by directly amplifying tangential components of estimated scores without modifying the model architecture.

Source code: https://github.com/hyeon-cho/Tangential-Amplifying-Guidance. Project page: https://hyeon-cho.github.io/TAG/

Examples (Flux):

TAG-Comparis

Closes https://github.com/comfyanonymous/ComfyUI/issues/10323.

### CODE REVIEW

**Comment by chaObserv** [comment]
This doesn’t seem like TAG. TAG splits the solver’s updates (DDIM, DPM, etc.) into normal and tangential components. Since it runs after a sampling step, the post-CFG level probably cannot handle this.

**Comment by kabachuha** [comment]
Well, I used the code straight from the repo: https://github.com/hyeon-cho/Tangential-Amplifying-Guidance/blob/74005bdd265c0a6d85099ab234b22b437ff6c774/pipelines/pipeline_tag_stablediffusion3.py#L375-L396

```python
                latents_dtype = latents.dtype
                output = self.scheduler.step(noise_pred, t, latents, return_dict=False)[0]
                if t <= sta_tpd and t >= end_tpd:
                    post_latents = latents 
                    v_t_2d       = post_latents / (post_latents.norm(p=2, dim=(1,2,3), keepdim=True) + 1e-8)

                    latents = output 

                    delta_latents = latents - post_latents
                    delta_unit    = (delta_latents * v_t_2d).sum(dim=(1,2,3), keepdim=True)

                    normal_update_vector     = delta_unit * v_t_2d
                    tangential_update_vector = delta_latents - normal_update_vector

                    eta_v = t_guidance_scale
                    eta_n = r_guidance_scale

                    latents = post_latents + \
                        eta_v * tangential_update_vector + \
                        eta_n * normal_update_vector
                else: # [NOTE] Simple Path (equal to original)
                    latents = output
```

I don't see it requires extra steps.

---

"sampler_post_cfg_function" is the end function of all samplers, basically just after the sampler's step technically 

https://github.com/comfyanonymous/ComfyUI/blob/b4f30bd4087a79b4c4fc89bb67b9889adb866294/comfy/samplers.py#L353-L390

**Comment by chaObserv** [comment]
```
output = self.scheduler.step(noise_pred, t, latents, return_dict=False)[0]
```
In diffusers, a scheduler's `step` function performs a single sampling step. For example, the `step` function in `EulerDiscreteScheduler` executes one Euler update.

Simplified Euler in comfy (k-diffusion) as example:
```
denoised = model(x, sigmas[i] * s_in, **extra_args)  # (post-CFG already applied)
d = to_d(x, sigmas[i], denoised)
dt = sigmas[i + 1] - sigmas[i]
x_next = x + d * dt

# TAG
sampler_increment = x_next - x
```
The post-CFG functions occur during the call to `model()`, so the resulting `denoised` is returned to the sampler function eventually. TAG is designed to amplify the tangential components of the update increment. However, the current implementation likely does not do this.
